frontend/src/hooks/useConnectWallet.js

```diff
@@ -0,0 +1,29 @@
+import { useMutation } from '@tanstack/react-query';
+import { connectWallet, checkForCRMToken } from 'services/wallet';
+import { notifySuccess, notifyError } from 'utils/notification';
+
+export const useConnectWallet = (setWalletId) => {
+  return useMutation({
+    mutationFn: async () => {
+     const walletAddress = await connectWallet();
+
+      if (!walletAddress) {
+        throw new Error('Failed to connect wallet');
+      }
+    const hasCRMToken = await checkForCRMToken(walletAddress);
+      if (!hasCRMToken) {
+        throw new Error('Wallet does not have CRM token');
+      }
+
+      return walletAddress;
+    },
+    onSuccess: (walletAddress) => {
+      setWalletId(walletAddress);
+      notifySuccess('Wallet successfully connected');
```

Comment:
Now I realise that we probably don't need success notification, remove it and just keep error one, thanks!

Reply:
noted